lib/data/locationConstraintParser.js

```diff
@@ -31,7 +32,38 @@ export default function parseLC() {
             clients[location].clientType = 'scality';
         }
         if (locationObj.type === 'aws_s3') {
-            // TODO
+            // users can either include the desired profile name from their
+            // ~/.aws/credentials file or include the accessKeyId and
+            // secretAccessKey directly in the locationConfig
+            if (locationObj.details.credentialsProfile) {
+                clients[location] = new AWS.S3({
+                    endpoint: `https://${locationObj.details.awsEndpoint}`,
+                    // Non-file stream objects are not supported
+                    // with SigV4 (node sdk)
+                    // so note that we are using the default of
+                    // signatureVersion v2
+
+                    // consider disabling
+                    debug: true,
+                    // perhaps use this rather than setting ourselves.
+                    // Not implemented yet for streams in node sdk!!!
+                    computeChecksums: true,
+                    credentials: new AWS.SharedIniFileCredentials({ profile:
+                        locationObj.details.credentialsProfile }),
+                });
+            } else {
+                clients[location] = new AWS.S3({
+                    endpoint: `https://${locationObj.details.awsEndpoint}`,
+                    debug: true,
+                    computeChecksums: true,
+                    accessKeyId: locationObj.details.credentials.accessKey,
+                    secretAccessKey: locationObj.details.credentials.secretKey,
+                });
+            }
+            clients[location].clientType = 'aws_s3';
+            clients[location].awsBucketName = locationObj.details.bucketName;
+            clients[location].bucketMatch = locationObj.details.bucketMatch;
+            clients[location].dataStoreName = location;
         }
```

Comment:
yup, they're asserted in Config.js. If the config values aren't right, the server throws an error and doesn't start